aws/publisher_test.go

```diff
@@ -91,3 +91,32 @@ func TestPublish(t *testing.T) {
 		}
 	}
 }
+
+// TestDeleteTopic verifies that a sns topic can be deleted successfully with the DeleteTopic function
+func TestDeleteTopic(t *testing.T) {
+	snsMock := mockSNS{}
+	p, errPub := newPublisher(&snsMock, "foo")
+	if errPub != nil {
+		t.Errorf("expected no error from newPublisher, but got: %v", errPub)
+	}
+
+	snsMock.stubbedDeleteTopicError = errors.New("test error")
+	expected := p.DeleteTopic()
```

Comment:
this is `actual`, not expected. The `expected` result is the one you've stubbed in the line above